<launch>

	<arg name="connected_to_baxter" default="false" />
	<arg name="use_webcam" default="false" />
	<arg name="use_kinect" default="false" />
	
	
	<arg name="kinect_img_in" default="/camera/rgb/image_color" />
	<!--arg name="uvccam_hydro_img_in" default="/camera/image_raw" /-->
	<arg name="loaded_img" default="/static_image" />
	<arg name="usbcam_img_in" default="/usb_cam/image_raw" />
	<arg name="baxter_L_img_in" default="/cameras/left_hand_camera/image" />
	<arg name="baxter_R_img_in" default="/cameras/right_hand_camera/image" />
	<arg name="baxter_H_img_in" default="/cameras/head_camera/image" />
	<arg name="rszd_img_w" default="160" />
	<arg name="rszd_img_h" default="100" />
	<arg name="scene_label_format" default="scene%01d" />
	<arg name="scene_ext" default=".JPG" />
	<arg name="scene_dir" default="/home/ros/scene_snap/sequence1/" />
	
	
	<!-- KINECT CAM -->
	<include if="$(arg use_kinect)" file="$(find openni_launch)/launch/openni.launch" />

	<!-- USB CAM -->
	<node if="$(arg use_webcam)" name="usb_cam" pkg="usb_cam" type="usb_cam_node" >
		<param name="video_device" value="/dev/video0" />
		<param name="image_width" value="640" />
		<param name="image_height" value="480" />
		<param name="pixel_format" value="mjpeg" />
		<param name="camera_frame_id" value="usb_cam" />
		<param name="io_method" value="mmap"/>
	</node>

	<!-- IMG LOADER -->
	<node unless="$(arg connected_to_baxter)" pkg="visor" type="imgloader" name="imgloader" required="true" output="screen">
		<param name="scene_label_format" value="$(arg scene_label_format)" />
		<param name="scene_ext" value="$(arg scene_ext)" />
		<param name="scene_dir" value="$(arg scene_dir)" />
	</node>

	<!-- IMG RESIZER (use loaded_img with the imgloader node) -->
	<node pkg="visor" type="img_resizer_node" name="img_resizer">
		<param if="$(arg connected_to_baxter)" name="camera_base_topic" value="$(arg baxter_L_img_in)" />
		<param unless="$(arg connected_to_baxter)" name="camera_base_topic" value="$(arg loaded_img)" />
		<param if="$(arg use_webcam)" name="camera_base_topic" value="$(arg usbcam_img_in)" />
		<param name="resized_width" value="$(arg rszd_img_w)" />
		<param name="resized_height" value="$(arg rszd_img_h)" />
		<param name="convert_to_bw" value="false" />
	</node>

</launch>
